src/yunohost/tools.py

```diff
@@ -162,6 +162,45 @@ def tools_maindomain(auth, new_domain=None):
         logger.warning("%s" % e, exc_info=1)
         raise MoulinetteError(errno.EPERM, m18n.n('maindomain_change_failed'))
 
+    # Set hostname
+    
+    commands = [     
+        "sudo hostnamectl --static set-hostname %s" % new_domain,
+        "sudo hostnamectl --transient set-hostname %s" % new_domain,
+        "sudo hostnamectl --pretty set-hostname (YunoHost/%s)" % new_domain
```

Comment:
I tried before committing, and somehow the quotes ended up in the name itself (in hostnamectl status). This probably has to do with the use of subprocess. Subprocess takes as input a list and each element kind of act like an argument. Not like bash where you put quotes specifically to say "this is to be understood as a single argument" (AFAIK).